rclpy/rclpy/node.py

```diff
@@ -1649,8 +1649,10 @@ def destroy_rate(self, rate: Rate):
 
         :return: ``True`` if successful, ``False`` otherwise.
         """
-        self.destroy_timer(rate._timer)
-        rate.destroy()
+        success = self.destroy_timer(rate._timer)
+        if success:
```

Comment:
I think `rate.destroy()` still needs to be called regardless of the result of destroying the timer. That will wake any threads sleeping on the `Rate`. Timers are removed from the node's `self._timers` list even when destroying them fails, so there isn't a second opportunity to destroy the rate instance and wake those threads.

Reply:
Fixed.